qiskit/transpiler/passes/basis/basis_translator.py

```diff
@@ -554,3 +607,13 @@ def _compose_transforms(basis_transforms, source_basis, source_dag):
                 )
 
     return mapped_instrs
+
+
+def _get_example_gates(source_dag, example_gates=None):
+    for node in source_dag.op_nodes():
+        example_gates[(node.op.name, node.op.num_qubits)] = node.op
+        if isinstance(node.op, ControlFlowOp):
+            for block in node.op.blocks:
+                block_dag = circuit_to_dag(block)
+                return {**example_gates, **_get_example_gates(block_dag, example_gates)}
+    return example_gates
```

Comment:
This recursion ends up being quadratic/exponential/some non-linear space complexity because of the reconstruction of the dictionary literals.  Something like this (again, cribbed from functional-programming styles) might work better

```python
def _get_example_gates(source_dag):
    def recurse(dag, gates=None):
        gates = gates or {}
        for node in dag.op_nodes():
            example_gates[(node.op.name, node.op.num_qubits)] = node.op
            if isinstance(node.op, ControlFlowOp):
                for block in node.op.blocks:
                    gates = recurse(circuit_to_dag(block), gates)
        return gates

    return recurse(source_dag)
```